x/config/config.go

```diff
@@ -60,14 +61,62 @@ func WrapStreamDialer(dialer transport.StreamDialer, transportConfig string) (tr
 	}
 	var err error
 	for _, part := range strings.Split(transportConfig, "|") {
-		dialer, err = newStreamDialerFromPart(dialer, part)
+		dialer.StreamDialer, err = newStreamDialerFromPart(dialer.StreamDialer, part)
+		dialer.SanitizedConfig = dialer.SanitizeConfig(part)
 		if err != nil {
 			return nil, err
 		}
 	}
 	return dialer, nil
 }
 
+type StreamDialer struct {
+	transport.StreamDialer
+	SanitizedConfig string
+}
+
+const redactedPlaceholder = "REDACTED"
+
+// SanitizeConfig returns a sanitized version of the given config that could contain sensitive information
+func (sd *StreamDialer) SanitizeConfig(oneDialerConfig string) string {
```

Comment:
Make it a standalone method, which is easier to reason about than a method that depends on internal state:
```suggestion
func sanitizeConfig(oneDialerConfig string) string {
```

This also follows the Single Responsibility Principle.